subsys/bluetooth/host/ecc.c

```diff
@@ -77,7 +88,7 @@ int bt_pub_key_gen(struct bt_pub_key_cb *new_cb)
 	if (err) {
 		BT_ERR("Sending LE P256 Public Key command failed");
 		atomic_clear_bit(bt_dev.flags, BT_DEV_PUB_KEY_BUSY);
-		pub_key_cb = NULL;
+		sys_slist_init(&pub_key_cb_slist);
```

Comment:
Should probably call each callback with a NULL pointer to avoid dropping callbacks silently